modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/tree/TrinomialTree.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.tree;
+
+import com.opengamma.strata.basics.value.ValueDerivatives;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+
+/**
+ * Trinomial tree.
+ * <p>
+ * Option pricing model based on trinomial tree. Trinomial lattice is defined by {@code LatticeSpecification} 
+ * and the option to price is specified by {@code OptionFunction}. 
+ * <p>
+ * Option pricing with non-uniform tree is realised by specifying {@code RecombiningTrinomialTreeData}.
+ */
+public class TrinomialTree {
+
+  /**
+   * Price an option under the specified trinomial lattice.
+   * <p>
+   * It is assumed that the volatility, interest rate and continuous dividend rate are constant 
+   * over the lifetime of the option. 
+   * 
+   * @param lattice  the lattice specification
+   * @param function  the option
+   * @param spot  the spot
+   * @param volatility  the volatility
+   * @param interestRate  the interest rate
+   * @param dividendRate  the dividend rate
+   * @return the option price
+   */
+  public double optionPrice(
+      LatticeSpecification lattice,
+      OptionFunction function,
```

Comment:
Make OptionFunction the first argument, like in other pricer and the other method in the same class.